src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/gadgets/MultiTool.java

```diff
@@ -73,25 +78,49 @@ protected ItemUseHandler getItemUseHandler() {
         return e -> {
             Player p = e.getPlayer();
             ItemStack item = e.getItem();
+            PersistentDataContainer container = item.getItemMeta().getPersistentDataContainer();
             e.cancel();
-
-            int index = selectedMode.getOrDefault(p.getUniqueId(), 0);
-
-            if (!p.isSneaking()) {
-                if (removeItemCharge(item, COST)) {
-                    SlimefunItem sfItem = modes.get(index).getItem();
-
-                    if (sfItem != null) {
-                        sfItem.callItemHandler(ItemUseHandler.class, handler -> handler.onRightClick(e));
+            
+            int index = 0;
+            if (container.has(key, PersistentDataType.STRING)) {
+                for (int i = 0; i < modes.size(); i++) {
+                    if (modes.get(i).getItemId() == container.get(key, PersistentDataType.STRING)) {
+                        index = i;
+                        break;
                     }
                 }
+            }
+
+            SlimefunItem sfItem = modes.get(index).getItem();
+
+            if (!p.isSneaking() && removeItemCharge(item, COST) && sfItem != null) {
+                sfItem.callItemHandler(ItemUseHandler.class, handler -> handler.onRightClick(e));
             } else {
                 index = nextIndex(index);
 
                 SlimefunItem selectedItem = modes.get(index).getItem();
                 String itemName = selectedItem != null ? selectedItem.getItemName() : "Unknown";
                 Slimefun.getLocalization().sendMessage(p, "messages.multi-tool.mode-change", true, msg -> msg.replace("%device%", "Multi Tool").replace("%mode%", ChatColor.stripColor(itemName)));
-                selectedMode.put(p.getUniqueId(), index);
+
+                ItemMeta meta = item.getItemMeta();
+                List<String> lore = meta.hasLore() ? meta.getLore() : new ArrayList<>();
+
+                meta.getPersistentDataContainer().set(key, PersistentDataType.STRING, modes.get(index).getItemId());
+
+                for (int i = 0; i < lore.size(); i++) {
+                    String line = lore.get(i);
+
+                    if (REGEX.matcher(line).matches()) {
+                        lore.set(i, LORE_PREFIX + ChatColor.stripColor(itemName));
+                        meta.setLore(lore);
+                        item.setItemMeta(meta);
+                        return;
+                    }
+                }
+
+                lore.add(2, LORE_PREFIX + ChatColor.stripColor(itemName));
+                meta.setLore(lore);
+                item.setItemMeta(meta);
```

Comment:
I feel like this can be abstracted or just done in a better way potentially, not entirely sure

Reply:
I thought so too, but I'm out of ideas for now.

Reply:
Nevermind, it was simpler than I thought. Fixed in https://github.com/Slimefun/Slimefun4/pull/4073/commits/c220609fd2e3b8fe25e99f33c918b9c009c02b92.